doc/nrf/libraries/dfu/dfu_extra.rst

```diff
@@ -0,0 +1,107 @@
+.. _lib_dfu_extra:
+
+DFU extra image extensions
+##########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The DFU extra image module provides a flexible mechanism for including extra firmware images in the DFU packages.
+This allows applications to extend the built-in DFU functionality with additional extra firmware images beyond ones supported natively in the nRF Connect SDK, for example firmware for external device.
+
+This CMake extension supports the following DFU delivery mechanisms:
+
+* **Multi-image binary** (``dfu_multi_image.bin``)
+
+  Multi-image binary is used by applications that require downloading the DFU package containing multiple images as a single blob. The application is then responsible for handling the images. Refer to :ref:`lib_dfu_multi_image` to see how to provide firmware implementation.
+
+* **ZIP packages** (``dfu_application.zip``)
+
+  ZIP packages are used for updates via SMP protocol. The firmware responsible for handling the images uses SMP image management which automatically handles DFU of up to 3 images (MCUboot slots). In case addition of yet another image is needed, user needs to adjust the ``zephyr_img_mgmt.c`` file. For more information about SMP, see :ref:`Device Management <zephyr:device_mgmt>` documentation.
+
+Configuration
+*************
+
+To enable extra image support, set the :kconfig:option:`SB_CONFIG_DFU_EXTRA_BINARIES` Kconfig option.
+
+Usage Examples
+**************
+
+Add extra binaries in your application's **sysbuild.cmake** file:
+
+.. code-block:: cmake
+
+   if(SB_CONFIG_DFU_EXTRA_BINARIES)
+      # Include the extra DFU system
+      include(${ZEPHYR_NRF_MODULE_DIR}/cmake/dfu_extra.cmake)
+
+      # Define path to external firmware
+      set(ext_fw "${APP_DIR}/ext_fw.signed.bin")
+
+      # Create build target for the external binary
+      add_custom_target(ext_fw_target
+         DEPENDS ${ext_fw}
+         COMMENT "External firmware dependency"
+      )
+
+      # Add extra binary to DFU packages
+      dfu_extra_add_binary(
+          IMAGE_ID 1
+          BINARY_PATH ${ext_fw}
+          DEPENDS ext_fw_target
+       )
+   endif()
+
+In case additional binary needs to be handled by MCUboot as updatable slots, adjust the :kconfig:option:`SB_CONFIG_MCUBOOT_UPDATEABLE_IMAGES` and make sure MCUboot partitions are defined correctly.
```

Comment:
```suggestion
In case additional binary needs to be handled by MCUboot as updatable slots, adjust the :kconfig:option:`SB_CONFIG_MCUBOOT_UPDATEABLE_IMAGES` Kconfig option, and make sure MCUboot partitions are defined correctly.
```